provider/pkg/schema/gen.go

```diff
@@ -1043,6 +1043,35 @@ func (ctx *context) propertyTypeSpec(parentName string, propSchema *jsschema.Sch
 		if !ok {
 			return nil, errors.Errorf("definition %s not found in resource %s", schemaName, ctx.cfTypeName)
 		}
+		var mapType *jsschema.Schema
+		if typeSchema.AdditionalProperties != nil && typeSchema.AdditionalProperties.Schema != nil {
+			mapType = typeSchema.AdditionalProperties.Schema
+		} else if typeSchema.PatternProperties != nil && len(typeSchema.PatternProperties) == 1 {
+			// TODO: Capture pattern add add to documentation or even provider metadata for early validation feedback.
+			// https://github.com/pulumi/pulumi-aws-native/issues/1346
+			for _, schema := range typeSchema.PatternProperties {
+				mapType = schema
```

Comment:
This is just the last of the patterns? Not sure I understand what's happening here.

Reply:
We're only looking at this if there's only a single pattern property in the map so we're just accessing the single value in the map here. It's not a list so I can't just do `typeSchema.PatternProperties[0]`.